### PR TITLE
feat(framework) Update context registration when running an app directory

### PR DESCRIPTION
**Pull request by jafermarq**
When there is not a `.fab` we still want to be able to read the config in the `pyproject.toml` and potentially override some values embedded in `<Run>.override_config`. To do this we need to call `get_fused_config_from_dir()` when registering a context.

### CODE REVIEW

File: src/py/flwr/client/node_state.py [hunk 1]
```diff
@@ -49,10 +49,22 @@ def register_context(
         run_id: int,
         run: Optional[Run] = None,
         flwr_dir: Optional[Path] = None,
+        fab_dir: Optional[str] = None,
```
Comment (danieljanes) [inline src/py/flwr/client/node_state.py:52]:
```suggestion
        app_dir: Optional[str] = None,
```

File: src/py/flwr/client/node_state.py [hunk 2]
```diff
@@ -49,10 +49,22 @@ def register_context(
         run_id: int,
         run: Optional[Run] = None,
         flwr_dir: Optional[Path] = None,
```
Comment (danieljanes) [inline src/py/flwr/client/node_state.py:51]:
```suggestion
        flwr_path: Optional[Path] = None,
```

File: src/py/flwr/client/node_state.py [hunk 3]
```diff
@@ -48,11 +48,23 @@ def register_context(
...
+        app_dir: Optional[str] = None,
     ) -> None:
         """Register new run context for this node."""
         if run_id not in self.run_infos:
-            initial_run_config = get_fused_config(run, flwr_dir) if run else {}
+            initial_run_config = {}
+            if app_dir:
+                # Load from FAB-like directory
+                app_path = Path(app_dir)
+                if app_path.is_dir():
```
Comment (danieljanes) [inline src/py/flwr/client/node_state.py:60]: Should we raise an error if this is `False`?

File: src/py/flwr/client/node_state.py [hunk 4]
```diff
@@ -48,11 +48,23 @@ def register_context(
...
-        flwr_dir: Optional[Path] = None,
+        flwr_path: Optional[Path] = None,
+        app_dir: Optional[str] = None,
     ) -> None:
         """Register new run context for this node."""
         if run_id not in self.run_infos:
-            initial_run_config = get_fused_config(run, flwr_dir) if run else {}
+            initial_run_config = {}
+            if app_dir:
+                # Load from FAB-like directory
```
Comment (danieljanes) [inline src/py/flwr/client/node_state.py:58]:
```suggestion
                # Load from app directory
```

File: src/py/flwr/client/node_state.py (hunk 3 above)
Comment (jafermarq) [reply src/py/flwr/client/node_state.py:60]: yes. done

**Review by danieljanes — APPROVED** [review]

**Pull request merged**